### PR TITLE
Fix duplicate headers

### PR DESCRIPTION
**Pull request by mmicael1**
https://github.com/dogecoin/dogecoin/issues/2021
Fix from https://github.com/bitcoin/bitcoin/pull/8054/files

### CODE REVIEW

**Comment by patricklodder** [comment]
I'm compiling this to deploy to some nodes in the wild. Thank you so much!

**Comment by mmicael1** [comment]
ping @patricklodder

**Comment by patricklodder** [comment]
> ping patricklodder

PONG! Still running tests.

**Comment by patricklodder** [comment]
TLDR; After weeks of puzzling, I unfortunately have to NAK this, because this creates a sync issue.

Here's why this is not a good patch (and the reversal in bitcoin/bitcoin - be it on gut feeling more than on hard data - was the right decision):

### Triggers for `getheaders` requests:

1. [Once, when connecting to a new peer, first thing after handshake](https://github.com/dogecoin/dogecoin/blob/97f95559c5ed8c986bcaee3252af5bf89e014f77/src/net_processing.cpp#L2918) 
2. [When we receive a `cmpctblock` message where we cannot seem to connect the block](https://github.com/dogecoin/dogecoin/blob/97f95559c5ed8c986bcaee3252af5bf89e014f77/src/net_processing.cpp#L1971)
3. [When we receive a `headers` message where we cannot seem to connect the header](https://github.com/dogecoin/dogecoin/blob/97f95559c5ed8c986bcaee3252af5bf89e014f77/src/net_processing.cpp#L2261)
4. [When we receive an `inv` message containing an unknown block](https://github.com/dogecoin/dogecoin/blob/97f95559c5ed8c986bcaee3252af5bf89e014f77/src/net_processing.cpp#L1563)
5. Before this patch: [When we receive a `headers` message containing 2000 records](https://github.com/dogecoin/dogecoin/blob/97f95559c5ed8c986bcaee3252af5bf89e014f77/src/net_processing.cpp#L2316)

### Problem with this patch

The condition introduced **_always_** evaluates to `false`, completely removing the 5th trigger, leaving nothing but the other 4 triggers to synchronize headers, which all are one-shot requests (because the only trigger that would fire more than once per external event is the fifth.)

If the condition were to evaluate to `true` based on the described behavior in the comment (which is not realized), we would limit ourselves to downloading headers from a single peer (the fastest) unless one of the other events would trigger, at which time we would re-evaluate peer speed and go with whomever reacts the fastest. Although that may be a better solution, it may not be a great solution either. I will look into alternatives.
